include/bulk/world.hpp

```diff
@@ -157,6 +166,10 @@ class world {
    */
   virtual std::unique_ptr<bulk::world> split(int part) = 0;
 
+  const std::unordered_map<std::string, uint64_t>& get_sync_counter() const {
```

Comment:
Is this needed for the application you have in mind? I would rather keep the `unordered_map` an implementation detail.